src/hotspot/share/opto/castnode.cpp

```diff
@@ -524,6 +524,12 @@ const Type* ConstraintCastNode::widen_type(const PhaseGVN* phase, const Type* re
   if (!phase->C->post_loop_opts_phase()) {
     return res;
   }
+
+  // Keep these casts for verification
```

Comment:
You could write about `VerifyConstraintCasts` with its 2 modes, and say why they are both useful.

Reply:
Done